docs/standard/threading/overview-of-synchronization-primitives.md

```diff
@@ -1,174 +1,167 @@
+
+Unlike <xref:System.Threading.Monitor>, the <xref:System.Threading.Mutex> class can be used for inter-process synchronization. To do that, use a named mutex, which is visible throughout the operating system. To create a named mutex instance, use a [Mutex constructor](<xref:System.Threading.Mutex.%23ctor%2A>) that specifies a name. You also can call the <xref:System.Threading.Mutex.OpenExisting%2A?displayProperty=nameWithType> method to open an existing named system mutex.
   
- Interlocked operations can be used with volatile memory guarantees to write applications that exhibit powerful non-blocking concurrency. However, they require sophisticated, low-level programming, so for most purposes, simple locks are a better choice.  
+For more information, see the [Mutexes](mutexes.md) article and the <xref:System.Threading.Mutex> API reference.
+
+### SpinLock struct
+
+The <xref:System.Threading.SpinLock?displayProperty=nameWithType> struct, like <xref:System.Threading.Monitor>, grants exclusive access to a shared resource. When <xref:System.Threading.SpinLock> attempts to acquire the taken lock, it waits in a loop repeatedly checking until the lock becomes available.
+
+For more information about benefits and drawbacks of using spin lock, see the [SpinLock](spinlock.md) article and the <xref:System.Threading.SpinLock> API reference.
+
+### ReaderWriterLockSlim class
+
+Use the <xref:System.Threading.ReaderWriterLockSlim?displayProperty=nameWithType> class to grant exclusive access to a shared resource for writing. Multiple threads can access the resource simultaneously for reading. You might want to use <xref:System.Threading.ReaderWriterLockSlim> to synchronize access to a shared data structure that supports thread-safe read operations, but requires exclusive access to perform write operation. When a thread requests exclusive access (for example, by calling the <xref:System.Threading.ReaderWriterLockSlim.EnterWriteLock%2A?displayProperty=nameWithType> method), subsequent reader requests block until all existing readers have exited the lock, and the writer has entered and exited the lock.
   
- For a conceptual overview, see [Interlocked Operations](../../../docs/standard/threading/interlocked-operations.md).  
+For more information, see the [Reader-Writer locks](reader-writer-locks.md) article and the <xref:System.Threading.ReaderWriterLockSlim> API reference.
+
+### Semaphore and SemaphoreSlim classes
+
+Use <xref:System.Threading.Semaphore?displayProperty=nameWithType> or <xref:System.Threading.SemaphoreSlim?displayProperty=nameWithType> to limit the number of threads that can access a resource concurrently. Additional threads that request the resource wait until any thread releases the semaphore.
```

Comment:
I mentioned this elsewhere, this is probably not the most common use case. Though come to think of it, it's not wrong either so maybe it's good as is. It depends how you look at it - it may be a single resource that tolerates a certain level of concurrency or multiple resources.

I suppose an alternative would be to just describe what a semaphore really is - a counter that is updated atomically, that "block" when it is 0 and so on. But I'm far from convinced that these docs are the place to describe general concepts such as semaphores, mutual exclusion etc.